File: python/Node.py
''' Node.py implements a Node . Node is an element of a linked list'''

class Node:
    ''' Single linked node'''
    ''' REF page 142'''
    def __init__(self,item, next):
        self.item = item
        self.next = next


    def GetItem(self):
        return self.item

    def GetNext(self):
        return self.next

    def SetItem(self,item):
        self.item = item

    def SetNext(self,next):
        self.next = next
    

class DoubleNode(Node):

    ''' DoubleNode inhirits from Node'''
    def __init__(self,item,next,previous):
        super().__init__(item,next)
        self.previous = previous

    def GetPrevious(self):
        return self.previous

    def SetPrevious(self,previous):
        self.previous = previous
        

File: cpp/Node.cpp
#include <cstddef>
#include "Node.hpp"



//  default contructor
Node::Node(): _item{0}, _next{nullptr}{

}

Node::Node(double value ): _item{value}, _next{nullptr}{

}


Node::Node(double value , Node * next_node): _item {value}, _next {next_node} {};

double Node::GetItem(){
    return _item;

}

void Node::SetItem(double value ){
    _item = value;
}

Node * Node::GetNext(){
    return _next;
}

void Node::SetNext(Node* new_node){
    _next= new_node;
}